server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskAction.java

```diff
@@ -119,15 +121,34 @@ private void checkPermission(@Nullable String projectUuid) {
     }
   }
 
-  private static void maskErrorStacktrace(Request wsRequest, CeActivityDto ceActivityDto) {
-    Set<AdditionalField> fromRequest = TaskAction.AdditionalField.getFromRequest(wsRequest);
-    if (!fromRequest.contains(TaskAction.AdditionalField.STACKTRACE)) {
+  private static void maskErrorStacktrace(CeActivityDto ceActivityDto, Set<AdditionalField> additionalFields) {
+    if (!additionalFields.contains(AdditionalField.STACKTRACE)) {
       ceActivityDto.setErrorStacktrace(null);
     }
   }
 
+  @CheckForNull
+  private String extractScannerContext(DbSession dbSession, String taskUuid, Set<AdditionalField> additionalFields) {
+    if (additionalFields.contains(AdditionalField.SCANNER_CONTEXT)) {
+      return dbClient.scannerContextDao().selectScannerContext(dbSession, taskUuid)
```

Comment:
The scanner context is never returned because you've set the taskUuid instead of the analysisUuid (so UT seems to be wrong :( )